met-web/src/components/engagement/admin/create/authoring/AuthoringSideNav.tsx

```diff
@@ -0,0 +1,281 @@
+import React from 'react';
+import {
+    ListItemButton,
+    List,
+    ListItem,
+    Box,
+    Drawer,
+    Toolbar,
+    SwipeableDrawer,
+    Grid,
+    Avatar,
+    ThemeProvider,
+} from '@mui/material';
+import { getAuthoringRoutes as getRoutes, AuthoringRoute as Route } from './AuthoringNavElements';
+import { DarkTheme, Palette, colors, ZIndex } from 'styles/Theme';
+import { AuthoringNavProps, DrawerBoxProps } from './types';
+import { When } from 'react-if';
+import { useAppSelector } from 'hooks';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faPencil } from '@fortawesome/pro-light-svg-icons/faPencil';
+import { Link } from 'components/common/Navigation';
+import { BodyText } from 'components/common/Typography';
+import { USER_ROLES } from 'services/userService/constants';
+import UserService from 'services/userService';
+import { faArrowRight } from '@fortawesome/free-solid-svg-icons';
+import { faArrowLeftLong } from '@fortawesome/pro-light-svg-icons';
+import { faCheck } from '@fortawesome/pro-regular-svg-icons';
+import { StatusCircle } from '../../view/AuthoringTab';
+
+export const routeItemStyle = {
+    padding: 0,
+    backgroundColor: Palette.background.default,
+    '&:hover, &:focus': {
+        filter: 'brightness(96%)',
+    },
+    '&:active': {
+        filter: 'brightness(92%)',
+    },
+    borderRadius: '8px',
+};
+
+const DrawerBox = ({ isMediumScreenOrLarger, setOpen, engagementId }: DrawerBoxProps) => {
+    const permissions = useAppSelector((state) => state.user.roles);
+    const tenant = useAppSelector((state) => state.tenant);
+
+    const currentBaseRoute = getRoutes(Number(engagementId), tenant)
+        .map((route) => route.base)
+        .filter((route) => location.pathname.includes(route))
+        .reduce((prev, curr) => (prev.length > curr.length ? prev : curr));
+
+    const allowedRoutes = getRoutes(Number(engagementId), tenant).filter((route) => {
+        return !route.authenticated || route.allowedRoles.some((role) => permissions.includes(role));
+    });
+
+    const renderListItem = (route: Route, itemType: string, key: number, engagementId: number) => {
```

Comment:
I feel like this could be a JSX element rather than a render function that returns an element - would allow us to use real JSX syntax within the `allowedRoutes.map()` call. It should also be extracted outside of the main DrawerBox component to avoid [typescript:S6478](https://sonarcloud.io/organizations/bcgov-sonarcloud/rules?open=typescript%3AS6478&rule_key=typescript%3AS6478)